src/configuration/vimrc.ts

```diff
@@ -0,0 +1,96 @@
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as path from 'path';
+import { IConfiguration, IKeyRemapping, IVimrcKeyRemapping } from './iconfiguration';
+import { vimrcKeyRemappingBuilder } from './vimrcKeyRemappingBuilder';
+
+class VimrcImpl {
+  public load(configuration: IConfiguration) {
+    if (configuration.vimrcPath) {
+      configuration.vimrcPath = VimrcImpl.expandHome(configuration.vimrcPath);
+      if (!fs.existsSync(configuration.vimrcPath)) {
+        return;
+      }
+    } else {
+      configuration.vimrcPath = VimrcImpl.findDefaultVimrc();
+      if (!configuration.vimrcPath) {
+        return;
+      }
+    }
+
+    let vimrcContent = fs.readFileSync(configuration.vimrcPath, { encoding: 'utf8' });
+    let lines = vimrcContent.split(/\r?\n/);
+
+    for (const line of lines) {
+      VimrcImpl.tryKeyRemapping(configuration, line);
+    }
+  }
+
+  private static tryKeyRemapping(configuration: IConfiguration, line: string): boolean {
+    let mapping: IVimrcKeyRemapping | null = vimrcKeyRemappingBuilder.build(line);
+    if (!mapping) {
+      return false;
+    }
+
+    let collection: IKeyRemapping[];
+    switch (mapping.keyRemappingType) {
+      case 'nmap':
+        collection = configuration.normalModeKeyBindings;
+        break;
+      case 'vmap':
+        collection = configuration.visualModeKeyBindings;
+        break;
+      case 'imap':
+        collection = configuration.insertModeKeyBindings;
+        break;
+      case 'nnoremap':
+        collection = configuration.normalModeKeyBindingsNonRecursive;
+        break;
+      case 'vnoremap':
+        collection = configuration.visualModeKeyBindingsNonRecursive;
+        break;
+      case 'inoremap':
+        collection = configuration.insertModeKeyBindingsNonRecursive;
+        break;
+      default:
+        return false;
+    }
+
+    // Don't override a mapping present in settings.json; those are more specific to VSCodeVim.
```

Comment:
Does this mean `set scrolloff=5` in my `.vimrc` won't override an unset `editor.cursorSurroundingLines`? Or only a *set* `editor.cursorSurroundingLines` will have higher priority?

Reply:
To be clear&mdash;currently, this PR only supports key remappings, and not other settings. 

The idea was that a user might want to bind the same key(s) differently within VS Code, _e.g._ so they could take advantage of VS Code's built-in commands. So, I called my new code after the remappings are read in from VS Code's settings, and checked to make sure a given key combination from the `.vimrc` was not present before adding it.

I wanted to support other settings as well, but for those I wasn't sure if it was possible to tell the difference between a default setting vs. a user setting. I was also uncertain about determining (automatically) which Vim settings are supported by VS Code Vim, or which might reasonably translate to built-in VS Code settings (like the new `editor.cursorSurroundingLines` you mentioned in another comment). It seemed like most of the people commenting in the issue were primarily concerned with key remappings, so I figured I would submit what I had, and perhaps and revisit the other settings later.

Reply:
Ah, gotcha. As long as we design with an eye forward (because if we support `.vimrc` at all, we'll be expected to support at least all the basics), that's fine, we can tackle it later.

I think we should probably maintain an explicit whitelist of supported settings, but these design decisions probably warrant a broader discussion.

Reply:
Resolving this chain because we'll handle other .vimrc features down the line - we're just doing remaps here.